docs: fix self-referential link in defer block description (#69952)
angular/angular: tools/manual_api_docs/blocks/defer.md

> A type of `block` that can be used to defer load the JavaScript for components,

Reviewer: ```suggestion
The `@defer` block can be used to lazy load components,
```
